### PR TITLE
Added Indic NLP News Classification tasks

### PR DESCRIPTION
**Pull request by Sakshamrzt**
- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+class IndicNLPNewsClassification(MultilingualTask, AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="IndicNLPNewsClassification",
+        dataset={
+            "path": "Sakshamrzt/IndicNLP-Multilingual",
+            "revision": "308230fabdcada106869a5196ba11c403d065b07",
+        },
+        description="A News classification dataset in multiple Indian regional languages.",
+        reference="https://github.com/AI4Bharat/indicnlp_corpus/blob/master/ai4bharat-indicnlp-corpus-2020.pdf",
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py:27]: We already have this dataset for Indian languages: [IndicSentimentClassification](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Classification/multilingual/IndicSentimentClassification.py). Is the dataset you're adding different from this one?

Comment (imenelydiaker) [reply mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py:27]: Okay, can you please add more information about the classes in the dataset card?

Comment (Sakshamrzt) [reply mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py:27]: Added

File: mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+class IndicNLPNewsClassification(MultilingualTask, AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="IndicNLPNewsClassification",
+        dataset={
+            "path": "Sakshamrzt/IndicNLP-Multilingual",
+            "revision": "3f23bd4a622a462adfb6989419cfadf7dc778f25",
+        },
+        description="A News classification dataset in multiple Indian regional languages.",
+        reference="https://github.com/AI4Bharat/indicnlp_corpus/blob/master/ai4bharat-indicnlp-corpus-2020.pdf",
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py:27]:
```suggestion
        reference="https://github.com/AI4Bharat/indicnlp_corpus#indicnlp-news-article-classification-dataset",
```
Maybe adding the llink to the dataset collection on GitHub is better

File: mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    def dataset_transform(self):
+        for lang in self.langs:
+            self.dataset[lang] = self.dataset[lang].rename_columns(
+                {"news": "text", "class": "label"}
+            )
+            if lang == "pa":
+                self.dataset[lang] = self.dataset[lang].remove_columns("headline")
+            self.dataset[lang] = self.stratified_subsampling(
+                self.dataset[lang], n_samples=256, seed=self.seed, splits=["test"]
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/multilingual/IndicNLPNewsClassification.py:61]: Why the number of samples is set to 256 ? We prefer subsampling to 2048 samples for the test set

File: results/intfloat__multilingual-e5-small/IndicNLPNewsClassification.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (imenelydiaker) [inline results/intfloat__multilingual-e5-small/IndicNLPNewsClassification.json:1]: I wasn't expecting results to be this high for low resource languages. Maybeit's due to the small number of samples selected in the subsampling function. Can you please run this again with 2048 samples instead?

**Review by imenelydiaker — COMMENTED** [review]
All good! Thanks for you contribution, please feel free to add points 🙂

File: docs/mmteb/points/610.jsonl [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "Sakshamrzt", "New dataset": 12, "Review PR": 4}
```
[the remark this answers is not in the document]
Comment (Sakshamrzt) [reply docs/mmteb/points/610.jsonl]: Yes. I have updated the points to just have the score for the current cl.

**Review by imenelydiaker — APPROVED** [review]
LGTM! Ready for merging, thank you for your contibution!

**Comment by KennethEnevoldsen** [comment]
re-enabled auto-merge on this PR

**Pull request merged**